### PR TITLE
Time out for communication.py

### PR DESCRIPTION
**Pull request by reiddewberry**
Added a pyserial timeout to return None when either we timeout or when we get bad data (which is determined by the json module throwing an exception because the data isn't valid json i.e. some little part got dropped and now json can't decode it).

### CODE REVIEW

File: control/communication.py [hunk 1]
```diff
@@ -17,8 +17,10 @@ def send(self, data):
         :param data: The data to send (must be able to be jsonned).
 
         """
+        #json data given by user and string encode it
```
Comment (halaney) [inline control/communication.py]: add space to pass linter checks ```# json``` instead of ```#json```

File: control/communication.py [hunk 2]
```diff
@@ -17,8 +17,10 @@ def send(self, data):
...
         jsoned = json.dumps(data)
         byte_data = str.encode(jsoned)
+        #send data
```
Comment (halaney) [inline control/communication.py]: same space deal

File: control/communication.py [hunk 3]
```diff
@@ -29,11 +31,17 @@ def receive(self):
...
-        :returns: original data -- This data has the type that was sent. It is
+        :returns:
+        None -- if the time_out time is exceeded
+
+        original data -- This data has the type that was sent. It is
                                    up to the user to determine the type and use
                                    it accordingly.
 
         """
+        #serial readline the data
```
Comment (halaney) [inline control/communication.py]: same space deal

File: control/communication.py [hunk 4]
```diff
@@ -29,11 +31,17 @@ def receive(self):
...
         jsoned_data = self.ser.readline()
+        if not jsoned_data:
+            return None
         unjsoned_data = json.loads(jsoned_data)
```
Comment (halaney) [inline control/communication.py]: I think you meant to put the try except thing here for handling json errors, right now if we get bad data it will throw and exception. you should catch it and return ```None``` if we get bad data

**Review by halaney — CHANGES_REQUESTED** [review]
Make sure this is passing: https://travis-ci.org/team-204/control/builds/362218429?utm_source=github_status&utm_medium=notification
1: Fix the linter errors
2: Add the try except block for when there is bad json data
3: Fix the unit tests that are failing (since the public api changed for the constructor)
4: Add a unit test for a timeout, and for when we get bad data, ensuring they both return None as expected

File: control/communication.py (hunk 4 above)
Comment (halaney) [reply control/communication.py]: Atleast that's what you made it seem like this is supposed to do from the comment you got at the top of this pull request

**Review by halaney — APPROVED** [review]
